exopy_hqc_legacy/tasks/tasks/instr/transfer_awg_file_task.py

```diff
@@ -0,0 +1,484 @@
+                self._pack_record('SEQUENCE_WAIT_{}'.format(i+1),
+                                  trig_wait[i], 'h') +
+                self._pack_record('SEQUENCE_LOOP_{}'.format(i+1),
+                                  int(nrep[i]), 'l') +
+                self._pack_record('SEQUENCE_JUMP_{}'.format(i+1),
+                                  jump_to[i], 'h') +
+                self._pack_record('SEQUENCE_GOTO_{}'.format(i+1),
+                                  goto_state[i], 'h'))
+            for ch_id in list(wfname_l.keys()):
+                wfname = wfname_l[ch_id][i]
+                seq_record_str.write(
+                    self._pack_record('SEQUENCE_WAVEFORM_NAME_CH_' + str(ch_id)
+                                      + '_{}'.format(i+1), wfname + '\x00',
+                                      '{}s'.format(len(wfname + '\x00')))
+                )
+
+        awg_file = (head_str.getvalue() + ch_record_str.getvalue() +
+                    wf_record_str.getvalue() + seq_record_str.getvalue())
+
+        return awg_file
+
+    def prepare_sequences(self, wait_trigger, start_with_event):
+        seq = self.sequence
+        context = seq.context
+
+        packed_waveforms = {}
+        wfname_l = {}
+        nrep = []
+        trig_wait = []
+        goto_state = []
+        jump_to = []
+
+        loops = []
+        name_parameters = []
+        n_loops = len(self.parameters)
+
+        first_index = 1
+        if n_loops > 0:
+            for params in self.parameters.items():
+                loop_start = float(self.format_and_eval_string(params[1][0]))
+                loop_stop = float(self.format_and_eval_string(params[1][1]))
+                loop_points = int(self.format_and_eval_string(params[1][2]))
+                loops.append(np.linspace(loop_start, loop_stop, loop_points))
+                name_parameters.append(params[0])
+                self.write_in_database(params[0]+'_loop',
+                                       np.linspace(loop_start,
+                                                   loop_stop,
+                                                   loop_points))
+
+            loop_values = np.moveaxis(np.array(np.meshgrid(*loops)), 0, -1)
+            loop_values = loop_values.reshape((-1, n_loops))
+        else:
+            loop_values = [1]
+        self.write_in_database('num_loop', len(loop_values))
+        for nn, loop_value in enumerate(loop_values):
+            if n_loops > 0:
+                for ii, name_parameter in enumerate(name_parameters):
+                    self.write_in_database(name_parameter, loop_value[ii])
+            for k, v in self.sequence_vars.items():
+                seq.external_vars[k] = self.format_and_eval_string(v)
+            context.sequence_name = '{}'.format(nn+first_index)
+            compilation = context.compile_loop(seq, for_file=True,
+                                               factor=False)
+            if len(compilation) == 5:
```

Comment:
This is ugly the method should be updated to always return the same number of objects.